pkg/operator/apiserver/controller/apiservice/apiservice_controller.go

```diff
@@ -23,8 +24,20 @@ import (
 	"github.com/openshift/library-go/pkg/operator/v1helpers"
 )
 
-type GetAPIServicesToMangeFunc func() ([]*apiregistrationv1.APIService, error)
-type apiServicesPreconditionFuncType func([]*apiregistrationv1.APIService) (bool, error)
+// ManagedAPIService keeps expected state of a single APIService object
+type ManagedAPIService struct {
+	// Enabled signals whether an APIService is enabled.
+	// Any disabled APIService is removed from a cluster.
+	Enabled bool
+	// APIService stores an actual APIService
+	APIService *apiregistrationv1.APIService
+}
+
+// GetAPIServicesToMangeFunc provides a full list of managed APIService items.
+// The list needs to always contain all the managed APIServices so only managed
+// objects are removed without removing user-created/unmanaged objects.
+type GetAPIServicesToMangeFunc func() ([]ManagedAPIService, error)
```

Comment:
why this function couldn't return services that are enabled :) ?

Reply:
how will you know if an api service is not enabled ?

Reply:
> why this function couldn't return services that are enabled :) ?

The controller needs a list of APIServices that are disabled. So they can be deleted if present.

> how will you know if an api service is not enabled ?

Configurable by anyone who consumes the APIService controller. Atm, the OA operator constructs the list based on enabled capabilities.